src/main/scala/io/citrine/lolo/bags/Bagger.scala

```diff
@@ -114,64 +114,23 @@ case class Bagger(
       .reduce(Bagger.combineImportance)
       .map(_.map(_ / importances.size))
 
-    /* Out-of-bag error and uncertainty for each point, calculating by combining the result of each tree.
-    Define as lazy so we only compute them if they are needed for the ratio or bias learner calculation */
-    lazy val oobErrors: Seq[(Vector[Any], Double, Double)] = trainingData.indices.flatMap { idx =>
-      val oobModels = models.zip(Nib.map(_ (idx))).filter(_._2 == 0).map(_._1).asInstanceOf[ParSeq[Model[PredictionResult[Double]]]]
-      if (oobModels.size < 2) {
-        None
-      } else {
-        Async.canStop()
-        val model = new BaggedModel(oobModels, Nib.filter {
-          _ (idx) == 0
-        }, useJackknife, disableBootstrap = disableBootstrap)
-        val predicted = model.transform(Seq(trainingData(idx)._1))
-        val error = predicted.getExpected().head - trainingData(idx)._2.asInstanceOf[Double]
-        val uncertainty = predicted match {
-          case x: RegressionResult => x.getStdDevObs.get.head
-          case _: Any => throw new UnsupportedOperationException("Computing oobErrors for classification is not supported.")
-        }
-        Some(trainingData(idx)._1, error, uncertainty)
-      }
-    }
-
-    /* Calculate the uncertainty calibration ratio, which is the 68th percentile of error/uncertainty
-    for the training points. If a point has 0 uncertainty, the ratio is 1 if error is also 0, otherwise infinity */
-    val ratio = if (uncertaintyCalibration && isRegression && useJackknife) {
-      Async.canStop()
-      oobErrors.map {
-        case (_, 0.0, 0.0) => 1.0
-        case (_, _, 0.0) => Double.PositiveInfinity
-        case (_, error, uncertainty) => Math.abs(error / uncertainty)
-      }.sorted.drop((oobErrors.size * 0.68).toInt).head
-    } else {
-      1.0
-    }
-    assert(!ratio.isNaN && !ratio.isInfinity, s"Uncertainty calibration ratio is not real: $ratio")
 
     /* Wrap the models in a BaggedModel object */
```

Comment:
I think we can avoid the nested conditionals and make this a bit cleaner as follows:

```
    /* Wrap the models in a BaggedModel object */
    val helper = BaggerHelper(models, trainingData, Nib, useJackknife, uncertaintyCalibration)
    val biasModel = if (biasLearner.isDefined && helper.oobErrors.nonEmpty && helper.isRegression) {
      Async.canStop()
      Some(biasLearner.get.train(helper.biasTraining).getModel().asInstanceOf[Model[PredictionResult[Double]]])
    } else {
      None
    }

    if (helper.isRegression) {
      new BaggedTrainingResult[Double](models.asInstanceOf[ParSeq[Model[PredictionResult[Double]]]], averageImportance, Nib, trainingData, useJackknife, biasModel, helper.ratio, disableBootstrap)
    } else {
      new BaggedTrainingResult[Any](models, averageImportance, Nib, trainingData, useJackknife, None, helper.ratio, disableBootstrap)
    }
```

That is how I tackled it when going to make this change. Also, we can remove the `isRegression` defined above in the method and just use `helper.isRegression`.

Reply:
yeah that's better